Use flex to automatically size optionContainer to its contents.

peacechen opened the pull request:
This allows the modal contents to automatically fit any screen.

peacechen committed:
Use flex to automatically size optionContainer to its contents. 

overlayStyle and cancelContainer adjusted accordingly to dynamically center content in a generic fashion.

fabriziomoscon commented:
merged into my fork

peacechen commented:
Thanks @fabriziomoscon. Do you publish your fork to NPM?

I noticed that you also removed flex:1 from selectStyle for RN 0.39 compatibility:
https://github.com/fabriziomoscon/react-native-modal-picker/commit/423c8a0395c0a2007cc47e99a6b39651439736b1

I haven't tested against 0.39, but if you've vetted it I'll add that change to my PR.

fabriziomoscon commented:
At the moment my `package.json` has this:

```
"react-native-modal-picker": "fabriziomoscon/react-native-modal-picker",
```

I am not planning to republish it under a different name.

Yes I tested it with RN 0.39.2 on android

peacechen committed:
remove flex:1 from selectStyle to make the CSS work with RN39. 

- versions of react native < 0.39.0 should pass flex:1 explicitly to selectStyle as a prop. 
- Update Readme to reflect selectStyle prop change for RN 0.39.

jaredtibs commented:
Is there an update on getting this merged? In need of this functionality but would like to use the original repo.

fabriziomoscon commented:
This repo is abandoned, I have cheery picked few PR from here in my fork since I am using this code in production.

peacechen committed:
Add prop keyboardShouldPersistTaps to pass to ScrollView.

peacechen committed:
Conditionally use ViewPropTypes.

peacechen committed:
Rename project

peacechen committed:
Update props in Readme

Commit:
Added NPM-version number and link to NPM-package (#3)

peacechen committed:
Update Readme

Commit:
Wrapping of TextInput works in iOS fixes #4 (#5)

Set the container prop `pointerEvents="none"` for children (content) to allow touches to `TextInput` to flow through.

Commit:
Add disabled prop to modal picker (#9)

peacechen committed:
Support orientation and resolution changes (#10)

Fixes sizing when rotating between landscape & portrait orientations.

Commit:
Prop to change supported orientation (#11)

* Prop to change supported orientation

fabriziomoscon committed:
Eslint rules (#15)

* add base eslint rules and fix lint errors
* add arbitrary eslint rules
- key-spacing
- no-empty-functions
- comma-dangle rules to prevent runtime errors

Commit:
Typo in prop-type declaration

Commit:
Removed react and react-native as devdependencies (#16)

peacechen committed:
Better way to support ViewPropTypes.

Backwards compatible with RN pre-0.44.

Commit:
Upgrade SampleApp to RN 0.47 (#8)

Commit:
Added prop for custom styling of options container (#21)

mvanroon committed:
add support for configurable animationType (#22)

* add support for configurable animationType

* updated documentation

peacechen committed:
Readme: Pretty props table

Update bitmap link and improve sample code.

peacechen committed:
Update version for npm release

Commit:
Add selectTextStyle to propTypes/defaultProps, closes #24

ChangedLater committed:
reverted style changes introduced in orientation fixes (#20)

* reverted style changes introduced in orientation fixes

* Updated style changes to correctly support cancel button

* reduced padding to 5%

* Updated overlayStyle in readme

Commit:
Update version for npm release to v0.0.19

mienaikoe committed:
set selectTextStyle propType to Text.propTypes.style (#26)

peacechen committed:
Update version for npm release

flyingace committed:
Optionally close the modal by pressing the overlay (#31)

* Optionally close the modal by pressing the overlay

In response to Close Modal on overlay click #28 https://github.com/peacechen/react-native-modal-selector/issues/28 I have made it possible to pass a prop to the component ('backdropPressToClose', boolean, default value is false) which when true will close the modal but calling the close() method, just as pressing the 'cancel' button in the modal does.

flyingace committed:
Added closeModalOnOverlayPress to documentation (#32)

* Added backdropPressToClose to documentation

peacechen committed:
Update version for npm release

Commit:
Upgrade SampleApp to RN 0.49 (#34)

Commit:
SampleApp: workaround for installation with yarn

Commit:
Fix onChange when modal closes (#35)

* Added onDismiss prop to Modal, available from RN >= 0.50

* Upgraded SampleApp to RN 0.50

terohotti committed:
remove bottom border from last item (#38)

Commit:
Update npm version to 0.0.22

Commit:
Cleaned up SampleApp

Commit:
Bugfix: Android does not have onDismiss prop

Commit:
Check for onDismiss-prop in a way that works on Expo

Commit:
Version 0.0.23, fixed android-bug introduced in v0.0.22

antonsivogrivov committed:
add cancelContainerStyle prop (#41)

* add cancelContainerStyle
* Remove flexGrow & maxHeight from the `cancelContainer` style

Commit:
Update NPM-version to v0.0.24

peacechen committed:
Add `cancelContainerStyle` prop to Readme

Commit:
Use arrow instead of bind (#40)

* Use arrow functions instead of manually bind methods

* Remove unused state variable

* Use proptypes from Modal-library instead of creating own ones

* Set initial state in constructor, not in componentDidMount

chriscohoat committed:
Added accessibility labels (#49)

* Add accessibility props

eduardojunio committed:
childrenContainerStyle and touchableStyle props added (#51)

* childrenContainerStyle and touchableStyle props added

anovoselnik committed:
Add test id prop (#48)

* add passThruProps prop for container View and options' TouchableOpacity.
* may be used to pass through testId or any desired custom prop.
* update README props section.

bdelville committed:
Key/Label extractors of data (#56)

* Add Key/Label extractors of data

Commit:
#44 Add TouchableOpacity activeOpacity option via props (#53)

* #44 Add TouchableOpacity activeOpacity option via props

Commit:
Update NPM-version to v0.0.26

sergey-kozel committed:
Add modal opening/closing event callbacks (#59)

peacechen committed:
Replace componentWillReceiveProps with componentDidUpdate

Fixes warning from RN 0.54+ / React 16+
See #60

Commit:
Readme: Fix formatting of props and code example

peacechen committed:
Add `visible` prop to control open/close state of modal.

See #46

peacechen committed:
Update NPM-version to v0.0.27

eslint: allow empty arrow functions

MattyK14 committed:
Add modalOpenerHitSlop prop (#64)

* Add the modalOpenerHitSlop prop to the TouchableOpacity that opens the modal selector.
https://facebook.github.io/react-native/docs/touchablewithoutfeedback.html#hitslop
* Documented in Readme

MattyK14 committed:
Change default modalOpenerHitSlop object (#66)

Set default hitSlop values. Empty object caused some unexpected & inconsistent behaviour.

jhellman committed:
Disable accessibility for modal TouchableWithoutFeedback (#73)

Disable accessibility for the TouchableWithoutFeeback backdrop (which dismisses the modal) so automation and users with accessibility features enabled can interact with the option list.

LasseSLambertsen committed:
Added closeOnChange prop (#74)

* Added closeOnChange prop
Useful if you want to implement a confirmation prompt after selection.

Jyrno42 committed:
Add prop customSelector (#72)

The prop can be used to render a custom node instead of the built-in select box.

Note: If using this prop one must control opening/closing themselves (either via `visible` prop or by calling `open/close` on the ref).

mikaello committed:
Add prop for custom styling selected element (#82)

* Enabling custom styling of selected item with 'selectedItemTextStyle'

Closes #75

* Add 'selectedItemTextStyle' to README

mikaello committed:
Update NPM-version to v0.0.28

mikaello committed:
Upgrade SampleApp to RN 0.56 (#83)

mikaello committed:
Add custom component/selector example in README/SampleApp (#84)

sampsasaarela committed:
Add get changed item function (#81)

* added getChangedItem method

* wait state update before close on change

* wait state update before close on change (fixed syntax error)

* renamed getChangedItem to getSelectedItem

* added getSelectedItem method description

mikaello committed:
Update NPM-version to v0.0.29

jessamp committed:
Accessibility Flexibility (#89)

* Add more accessibility props: cancelButtonAccessible, scrollViewAccessible, listItemAccessible, openButtonContainerAccessible. 
* Updated docs for accessibility. Also fixed a problem with android when open the list, the first item should be focused now.

mikaello committed:
Update NPM-version to v0.0.30

mikaello committed:
Update NPM-version to v1.0.0.

VacuumDecay committed:
Added pass through props for Text components (#92)

* Add pass-through props for Select Text and Option Text components

peacechen committed:
Update SampleApp to RN 0.57.8

peacechen committed:
Add support for custom components in options list (#98).

Supported for both sections and options.
Update Readme and SampleApp.

peacechen committed:
Update NPM-version to v1.0.1

dominicrico committed:
Update for RN 0.58+ (#99)

* Update Modal propTypes due to deprecation in RN 0.58+

peacechen committed:
Update NPM-version to v1.0.2

JoseVf committed:
Add selectedKey prop (#102)

* Add selectedKey prop
* Update Readme

peacechen committed:
Update NPM-version to v1.0.3

Commit:
Add initValueTextStyle prop (#106)

* Add initValueTextStyle prop

mikaello committed:
Upgrade SampleApp to RN 0.59.3

360disrupt committed:
Update: Integrated Git-Start into Readme.md (#110)

Add Git-Start links to Readme

mikaello committed:
Update NPM-version to v1.1.0

JolivetteSax committed:
Only use TouchableWithoutFeedback if backdropPressToClose is set. 

Fixes Expo issue when displaying the modal where the TouchableWithoutFeedback was causing a problem due to a RN bug in older versions (#118)

peacechen committed:
Update NPM-version to v1.1.1

mmarinovic committed:
Typescript typings (#123)

* added type definitions

* fix space

* label extractor returns string

* label optional for data item

* added markdown to type summary, added correct type for key property

mikaello committed:
Update NPM-version to v1.1.2

hitchcocknz committed:
onModalClose returns selected item (#107)

* Add selected item to onModalClose function

sneljo1 committed:
Fix selectTextStyle should have correct TextStyle type (#131)

mikaello committed:
Update NPM version to v1.1.3

Fix in TypeScript-types

jasonwoan committed:
Update selected state when data is dirty (#132)

peacechen committed:
Update NPM version to v1.1.4

peacechen committed:
Convert SampleApp to CRNA for better iOS & Android support.

peacechen committed:
Add props to support long press (#136)

peacechen committed:
Update NPM version to v1.1.5

Fix Readme typo

jacobkring committed:
Call `onChange` unconditionally to fix callback for iOS (#97)

This may have a side effect of firing onChange twice on iOS.

peacechen committed:
Update NPM version to v2.0.0

Bump major version as a precaution for onChange changes for iOS (#139)

vicdup committed:
Fix importantForAccessibility props for RN062 (#142)

peacechen committed:
Update NPM version to v2.0.1

peacechen committed:
Fix accessibility label default props (#128). 

Change scrollViewAccessibilityLabel and cancelButtonAccessibilityLabel defaults to empty strings `""`.

peacechen committed:
SampleApp: support cross-platform cp.

peacechen committed:
Add prop `scrollViewPassThruProps` (#146).

peacechen committed:
Update NPM version to v2.0.2

Remove unnecessary lock files which can trigger Github security warnings.

Merlier committed:
Add prop `optionsTestIDPrefix` for test automation. Support `testID` key in data options (#148)

Co-authored-by: Merlier <merlier771@gmail.com>

tetianastepova committed:
Add ability to pass extra props to Cancel button text in modal (#149)

peacechen committed:
Update NPM version to v2.0.3

peacechen committed:
Add eslint dev dependencies

Commit:
Add `listType` prop to support FlatList or ScrollView (#157)

Co-authored-by: HP <himanshu.patil@bizotics.com>
Co-authored-by: Peace <github@ghpc.33mail.com>

peacechen committed:
Fix `renderFlatlistOption()` bug (should return renderOption result)

peacechen committed:
Readme: add `listType` prop

ruru-ink committed:
Fix prop types for web view in react native (#158) (#162)

Co-authored-by: Phil Taylor <phil@customd.com>

peacechen committed:
2.0.4

peacechen committed:
onChange should fire only for valid option with key (#140)

peacechen committed:
2.0.5

peacechen committed:
Fix corner case onChange when key is 0

peacechen committed:
2.0.6

peacechen committed:
Use keyExtractor in onChange() (#140)

peacechen committed:
2.0.7

scousino committed:
TS: Add option param to onModalClose type (#166, #140)

peacechen committed:
2.0.8

mikaello committed:
types: fix FlatList types

See #168 for description of error

scousino committed:
Export props (#173)

* add option param to onModalClose type

* export props

mikaello committed:
docs: cleanup and added docs for methods

Commit:
Expo update and type fixes (#161)

* update

1) updated expo version of sample app to 40.0.0
2) added typescript to sample app
3) converted sample app.tsx from class component to functional
4) fixed types issue

* added more types

mikaello committed:
v2.0.9

asdolo committed:
Add prop for rendering a header above the list

mikaello committed:
v2.1.0

mikaello committed:
chore: upgrade SampleApp to React 17

mikaello committed:
feat: add 'close()' to TypeScript typings

Resolves #176

mikaello committed:
chore: update babel / eslint

mikaello committed:
v2.1.1

ArthurEsterox committed:
Add `onEndReached` prop for FlatList (#180, #182)

Co-authored-by: Arthur Arakelyan <artur.araqelyan.2004@gmail.com>

peacechen committed:
2.1.2
